--- exit_from_boot2.S.dist	2024-06-23 02:35:44.855569106 +0300
+++ exit_from_boot2.S	2024-07-30 23:10:45.471535598 +0300
@@ -7,6 +7,10 @@
 #ifndef _BOOT2_HELPER_EXIT_FROM_BOOT2
 #define _BOOT2_HELPER_EXIT_FROM_BOOT2
 
+#ifndef CUSTOM_XIP_BASE
+#define CUSTOM_XIP_BASE (XIP_BASE + 0x100)
+#endif
+
 #include "hardware/regs/m0plus.h"
 
 // If entered from the bootrom, lr (which we earlier pushed) will be 0,
@@ -18,7 +22,7 @@
     beq vector_into_flash
     bx r0
 vector_into_flash:
-    ldr r0, =(XIP_BASE + 0x100)
+    ldr r0, =(CUSTOM_XIP_BASE)
     ldr r1, =(PPB_BASE + M0PLUS_VTOR_OFFSET)
     str r0, [r1]
     ldmia r0, {r0, r1}
